### PR TITLE
Suggestions

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: .github/workflows/auto-update-collector-version.yml [hunk 1]
```diff
@@ -8,12 +8,10 @@ on:
 jobs:
   build:
     name: Check collector version
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
```
Comment (trask) [inline .github/workflows/auto-update-collector-version.yml:16]: it doesn't look like other git tags/branches (other than `main`) are needed

File: scripts/update-collector-version.sh [hunk 2]
```diff
@@ -1,64 +1,43 @@
-#!/bin/bash
+#!/bin/bash -e
```
Comment (trask) [inline scripts/update-collector-version.sh:1]: I generally always use this, b/c it ensures script will fail fast and not end up doing weird stuff after something unexpected occurs

File: scripts/update-collector-version.sh [hunk 3]
```diff
@@ -1,64 +1,43 @@
...
+if ! command -v gh &>/dev/null; then
     echo "The command 'gh' is expected to exist and be configured in order to update to the latest otelcol."
     exit 1
 fi
 
-command -v jq 2>/dev/null
-if [ $? != 0 ]; then
-    echo "The command 'jq' is expected to exist."
-    exit 1
-fi
```
Comment (trask) [inline scripts/update-collector-version.sh:17]: I replaced `jq` below with `gh api -q`

File: scripts/update-collector-version.sh [hunk 4]
```diff
@@ -1,64 +1,43 @@
...
-
-while getopts d:c: flag
-do
-    case "${flag}" in
-        d) directory=${OPTARG};;
-        c) create_pr=${OPTARG};;
-    esac
-done
```
Comment (trask) [inline scripts/update-collector-version.sh:25]: these options didn't seem important for our usage

File: scripts/update-collector-version.sh [hunk 5]
```diff
@@ -1,64 +1,43 @@
...
 
 # here we don't like having different versions, so we skip creating the PR if contrib & core are not equal
 if [ $latest_core_version != $latest_contrib_version ]; then
     echo "The contrib and core versions aren't matching. This might be OK, but perhaps there's a release in process?"
-    exit -1
+    exit 1
 fi
 
-branch="auto-update/collector_version_${latest_core_version}"
+branch="opentelemetrybot/collector_version_${latest_core_version}"
```
Comment (trask) [inline scripts/update-collector-version.sh:22]: I've been naming `opentelemetrybot` branches this way (similar to `dependabot`)

also see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/repository-settings.md#dependabot-and-opentelemetrybot

File: scripts/update-collector-version.sh [hunk 6]
```diff
@@ -1,64 +1,43 @@
...
-if [[ $? == 0 ]]; then
+if git diff --quiet ${gettingStartedFile}; then
     echo "We are already at the latest versions."
     exit 0
 fi
 
 git add ${gettingStartedFile}
 
-git commit -sm "Bump OpenTelemetry collector version to ${latest_core_version}"
+git commit -m "Bump OpenTelemetry collector version to ${latest_core_version}"
```
Comment (trask) [inline scripts/update-collector-version.sh:37]: `-s` looks like its for signed commits which we don't need

**Comment by svrnm** [comment]
All of that looks good to me, thanks for the effort going through this @trask !

**Pull request merged**